ontoconv/pipelines.py

```diff
@@ -183,6 +196,155 @@ def generate_pipeline(
     }
 
 
+def generate_ontoflow_pipeline(
+    ts: Triplestore,
+    resources: dict,
+    recognised_keys: "Optional[Union[dict, str]]" = "basic",
+) -> dict:
+    """Return a declarative ExecFlow pipeline as a dict.
+
+    Arguments:
+        ts: Tripper triplestore documenting data sources and sinks.
+        resources: A dict referring to OTEAPI partial pipelines for
+            a set of data sources and sinks.
+            See the example below for the expected structure.
+        client_iri: IRI of OTELib client to use.
+
+    Returns:
+        Dict-representation of a declarative ExecFlow pipeline.
+
+    Examples:
+        Example of the `resources` argument:
+
+        ```python
+        {'sinks': [
+           {'iri': 'ss3:AbaqusConfiguration',
+            'resourcetype': 'ss3:AbaqusSimulation'},
+           {'iri': 'ss3:AluminiumMaterialCard',
+            'resourcetype': 'ss3:AbaqusSimulation'},
+           {'iri': 'ss3:ConcreteMaterialCard',
+            'resourcetype': 'ss3:AbaqusSimulation'}],
+         'sources': [
+           {'iri': 'ss3kb:abaqus_config1', 'resourcetype': 'dataset'},
+           {'iri': 'ss3:AluminiumMaterialCard',
+            'resourcetype': 'ss3:MaterialCardGenerator'},
+           {'iri': 'ss3kb:abaqus_materialcard_concrete1',
+            'resourcetype': 'dataset'}]}
+        ```
+    """
+    names = {"input": [], "output": []}
```

Comment:
`names` can just be a list. We don't need to separate between input and output, since `lst` is already sorted with output coming before input.

More importantly, `names` and `strategies` should be corresponding, where each element in `names` is the name of the name of the corresponding element in `strategies`.

Reply:
I think it is easier to keep track of everythin thing correctly if we have them as separate lists. Names and strategies are corresponding as seen further down. They are populated at the same time. The order of the names list is imoprtant, that of strategies is not.

Reply:
But that will quickly lead to duplication of code. The reason to introduce `lst` with was to avoid that.